### PR TITLE
Rcal 954 Remove jump code from rcal tree

### LINKED ISSUE #1518: Remove jump code from rcal tree
_Issue [RCAL-954](https://jira.stsci.edu/browse/RCAL-954) was created on JIRA by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis):_

Remove jump code from code base and update the documentation,

Check ramp_fit step to make sure the jump step cannot be selected.

### PR DESCRIPTION
**Pull request by ddavis-stsci**
Resolves [RCAL-954](https://jira.stsci.edu/browse/RCAL-954)

Closes #1518

This PR removes the old jump code for even ramps and updates the documentation. 

## Tasks
- [x] **request a review from someone specific**, to avoid making the maintainers review every PR
- [ ] add a build milestone, i.e. `24Q4_B15` (use the [latest build](https://github.com/spacetelescope/romancal/milestones) if not sure)
- [x] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [x] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [ x] update or add relevant tests
  - [x] update relevant docstrings and / or `docs/` page
  - [x] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [x] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/romancal/wiki/How-to-resolve-JIRA-issues)

<details><summary>news fragment change types...</summary>

  - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
  - ``changes/<PR#>.docs.rst``
  - ``changes/<PR#>.stpipe.rst``
  - ``changes/<PR#>.associations.rst``
  - ``changes/<PR#>.scripts.rst``
  - ``changes/<PR#>.mosaic_pipeline.rst``
  - ``changes/<PR#>.patch_match.rst``

  ## steps
  - ``changes/<PR#>.dq_init.rst``
  - ``changes/<PR#>.saturation.rst``
  - ``changes/<PR#>.refpix.rst``
  - ``changes/<PR#>.linearity.rst``
  - ``changes/<PR#>.dark_current.rst``
  - ``changes/<PR#>.jump_detection.rst``
  - ``changes/<PR#>.ramp_fitting.rst``
  - ``changes/<PR#>.assign_wcs.rst``
  - ``changes/<PR#>.flatfield.rst``
  - ``changes/<PR#>.photom.rst``
  - ``changes/<PR#>.flux.rst``
  - ``changes/<PR#>.source_detection.rst``
  - ``changes/<PR#>.tweakreg.rst``
  - ``changes/<PR#>.skymatch.rst``
  - ``changes/<PR#>.outlier_detection.rst``
  - ``changes/<PR#>.resample.rst``
  - ``changes/<PR#>.source_catalog.rst``
</details>
The regression tests are at https://github.com/spacetelescope/RegressionTests/actions/runs/11972755984
the one failure is for the ALL SATURNATED case where the truth file will need to be updated for the removal of the 
cal_step 'jump"

### CODE REVIEW

File: docs/roman/pipeline_run.rst [hunk 1]
```diff
@@ -64,9 +64,9 @@ From the Python Prompt
 ------------------------------
 
 You can execute a pipeline or a step from within python by using the
-``call`` method of the class.
+``run`` method.
 
-The ``call`` method creates a new instance of the class and runs the pipeline or
+The ``run`` method creates a new instance of the class and runs the pipeline or
```
Comment (braingram) [inline docs/roman/pipeline_run.rst:69]: `run` doesn't create a new instance. `call` does.

File: docs/roman/stpipe/call_via_call.rst [hunk 2]
```diff
@@ -1,72 +0,0 @@
-.. _call_examples:
```
Comment (braingram) [inline docs/roman/stpipe/call_via_call.rst:1]: Would you restore this file?

I think there's some confusion over https://github.com/spacetelescope/romancal/pull/1499 That PR removed uses of `__call__` which is now deprecated in stpipe main. `call` is still the recommended way of using pipelines/steps.

File: changes/1534.general.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Remove the jump code for the depricated jump detection for even ramps and update the documentation
```
Comment (mairanteodoro) [inline changes/1534.general.rst:1]: Typo: `depricated -> deprecated`.

File: docs/roman/stpipe/index.rst [hunk 4]
```diff
@@ -13,7 +13,6 @@ For Users
    user_pipeline.rst
    user_logging.rst
    config_asdf.rst
-   call_via_call.rst
```
Comment (mairanteodoro) [inline docs/roman/stpipe/index.rst:16]: We might need to restore this file too, for the reasons pointed by @braingram regarding the use of `run` and `call`.

**Review by mairanteodoro — COMMENTED** [review]
Overall, looks good, @ddavis-stsci! However, I agree with @braingram that we should stick with the recommended `call` instead of `run`.

File: docs/roman/stpipe/index.rst (hunk 4 above)
Comment (ddavis-stsci) [reply docs/roman/stpipe/index.rst:16]: I think that is out of scope here and the call vs run rewrite is in a separate ticket

Comment (braingram) [reply docs/roman/stpipe/index.rst:16]: Would you elaborate on this? Specifically:
- What's the separate ticket?
- Is romancal planning to not use `Step.call`? That's news to me.

File: docs/roman/pipeline_run.rst [hunk 5]
```diff
@@ -63,22 +63,21 @@ Exit Status
...
 supplying a parameter file. Some examples are shown below.
 
 For the exposure pipeline and steps,
 
 ::
 
  from romancal.pipeline import ExposurePipeline
- result = ExposurePipeline.call('r0000101001001001001_01101_0001_WFI01_uncal.asdf')
+ elp = ExposurePipeline()
+ result = elp('r0000101001001001001_0001_wfi01_uncal.asdf')
```
Comment (braingram) [inline docs/roman/pipeline_run.rst:76]: This is using `__call__` which is deprecated in `stpipe`. Perhaps we can discuss these changes in tomorrows tag up.

Comment (ddavis-stsci) [reply docs/roman/pipeline_run.rst:83]: I'm not sure why this is relevant for removing the jump code?

Comment (braingram) [reply docs/roman/pipeline_run.rst:83]: I agree. Would you remove this and the other `call` vs `__call__` changes from this PR?

File: .gitignore [hunk 6]
```diff
@@ -57,6 +57,8 @@ distribute-*.tar.gz
 .tox
 .*.sw[op]
 *~
+#*
+.#*
```
Comment (braingram) [inline .gitignore:61]:
```suggestion
```
What files are these lines attempting to ignore?
The first one is a comment (`#*`) the second looks like it ignores (`.`) which is the current directory (already not included).

Comment (ddavis-stsci) [reply .gitignore:61]: These are some of the emacs save files that can be left if it crashes. Things like
#%2Aediff-merge%2A#hDh6cL#
and 
.#%2%2A#hDh6cL#

Comment (braingram) [reply .gitignore:61]: Thanks. Would you move those to a section with a comment `# emacs crash files`?

The first example `#%2Aediff-merge%2A#hDh6cL#` isn't ignored with the current items but is ignored if you update `#*` to `\#*`. Would you also make that change?

File: .gitignore [hunk 7]
```diff
@@ -57,6 +57,10 @@ distribute-*.tar.gz
 .tox
 .*.sw[op]
 *~
+
+# emacs save files
```
Comment (braingram) [inline .gitignore:61]: This section broke up the existing `# Other` section. Would you move it before `# Other`?

File: docs/roman/pipeline_run.rst [hunk 8]
```diff
@@ -70,22 +70,21 @@ Exit Status
 From the Python Prompt
 ------------------------------
 
-You can execute a pipeline or a step from within python by using the
-``call`` method of the class.
-
-The ``call`` method creates a new instance of the class and runs the pipeline or
-step. Optional parameter settings can be specified by via keyword arguments or
```
Comment (braingram) [inline docs/roman/pipeline_run.rst:77]: Please restore this section.

File: docs/roman/pipeline_run.rst [hunk 9]
```diff
@@ -70,22 +70,21 @@ Exit Status
...
 supplying a parameter file. Some examples are shown below.
 
 For the exposure pipeline and steps,
 
 ::
 
  from romancal.pipeline import ExposurePipeline
- result = ExposurePipeline.call('r0000101001001001001_01101_0001_WFI01_uncal.asdf')
+ elp = ExposurePipeline()
+ result = elp.call('r0000101001001001001_0001_wfi01_uncal.asdf')
```
Comment (braingram) [inline docs/roman/pipeline_run.rst:83]:
```suggestion
 elp = ExposurePipeline()
 result = elp.call('r0000101001001001001_0001_wfi01_uncal.asdf')
```
`call` is a class method and doesn't require an instance. Please restore the code from main.

File: docs/roman/pipeline_run.rst [hunk 10]
```diff
@@ -70,22 +70,21 @@ Exit Status
...
 
  from romancal.linearity import LinearityStep
- result = LinearityStep.call('r0000101001001001001_01101_0001_WFI01_uncal.asdf')
+ linearity =  LinearityStep()
+ result = linearity.call('r0000101001001001001_0001_wfi01_uncal.asdf')
```
Comment (braingram) [inline docs/roman/pipeline_run.rst:87]:
```suggestion
 linearity =  LinearityStep()
 result = linearity.call('r0000101001001001001_0001_wfi01_uncal.asdf')
```
Same here, please restore the code from main.

File: docs/roman/pipeline_run.rst [hunk 11]
```diff
@@ -95,13 +94,11 @@ For the mosaic level pipeline and steps,
 ::
 
  from romancal.pipeline import MosaicPipeline
- result = ExposurePipeline.call('r0000101001001001001_asn.json')
+ mosp = MosaicPipeline('r0000101001001001001_asn.json')
```
Comment (braingram) [inline docs/roman/pipeline_run.rst:97]:
```suggestion
result = MosaicPipeline.call('r0000101001001001001_asn.json')
```
Thanks for switching this to `MosaicPipeline`. I think from the text above that this should be a `call` as it's describing the pipeline results.

File: docs/roman/pipeline_run.rst [hunk 12]
```diff
@@ -95,13 +94,11 @@ For the mosaic level pipeline and steps,
...
 
  from romancal.skymatch import SkyMatchStep
- result = SkyMatchStep.call('r0000101001001001001_asn.json')
-
+ skymatch = SkyMatchStep.call('r0000101001001001001_asn.json')
```
Comment (braingram) [inline docs/roman/pipeline_run.rst:100]:
```suggestion
result = SkyMatchStep.call('r0000101001001001001_asn.json')
```
Assigning to `skymatch` is misleading as `call` is not returning a `SkyMatchStep` instance.

File: docs/roman/stpipe/call_via_run.rst [hunk 13]
```diff
@@ -6,8 +6,7 @@ Executing a pipeline or pipeline step directly, or via run()
 When calling a pipeline or step instance directly, or using the ``run`` method,
 you can specify individual parameter values manually. In this case, parameter
 files are not used. If you use ``run`` after instantiating with a parameter
-file (as is done when using the :ref:`call<call_examples>` method), the
-parameter file will be ignored.
```
Comment (braingram) [inline docs/roman/stpipe/call_via_run.rst:10]: Please restore these docs.

File: docs/roman/stpipe/config_asdf.rst [hunk 14]
```diff
@@ -173,14 +177,14 @@ step is executed.
 Remember that parameter values can come from numerous sources. Refer to
 :ref:`Parameter Precedence` for a full listing of how parameters can be set.
 
-From the ``JumpStep`` example, if all that needed to change is the
-``rejection_threshold`` parameter with a setting of ``80.0``,
+From the ``SourceCatalogStep`` example, if all that needed to change is the
+``aperture_ee1`` parameter with a setting of ``30.0``,
```
Comment (braingram) [inline docs/roman/stpipe/config_asdf.rst:181]:
```suggestion
``aperture_ee1`` parameter with a setting of ``30.0``,
```
`aperture_ee1` is an integer. Attempting to use `30.0` results in an error:
`ValueError: Config parameter 'aperture_ee1': the value "30.0" is of the wrong type.`. Please change this and the other uses to `30`.

File: docs/roman/stpipe/user_pipeline.rst [hunk 15]
```diff
@@ -154,39 +156,37 @@ equivalent of the parameter file above:
...
+    > strun romancal.pipeline.ExposurePipeline --steps.source_catalog.aperture_ee1=3
 
 From Python
 -----------
 
 Once the pipeline has been configured (as above) it can be executed
 using run.
 
-    pipe.run(input_data)
+    pipe(input_data)
```
Comment (braingram) [inline docs/roman/stpipe/user_pipeline.rst:189]:
```suggestion
    pipe.run(input_data)
```
Using `pipe(input_data)` executes `pipe.__call__` (which is deprecated). Please restore the example from main `pipe.run`.

File: docs/roman/stpipe/user_step.rst [hunk 16]
```diff
@@ -172,51 +172,16 @@ for ``strun``. The call signature is::
...
 the equivalent `from_cmdline` call would be::
 
     from romancal.pipeline import ExposurePipeline
-    ExposurePipeline.from_cmdline([' r0000101001001001001_01101_0001_WFI01_uncal.asdf',
-                                   'steps.jump.override_gain', 'roman_wfi_gain_0033.asdf'])
+    ExposurePipeline.from_cmdline([' r0000101001001001001_0001_wfi01_uncal.asdf',
+                                   'steps.ramp_fit.override_gain', 'roman_wfi_gain_0033.asdf'])
 
 
-call()
```
Comment (braingram) [inline docs/roman/stpipe/user_step.rst:185]: Please restore this section.

File: docs/roman/stpipe/user_step.rst [hunk 17]
```diff
@@ -233,11 +198,11 @@ example is::
 `input` in this case can be a asdf file containing the appropriate data, or the output
 of a previously run step/pipeline, which is an instance of a particular :ref:`datamodel<datamodels>`.
 
-Unlike the ``call`` class method, there is no parameter initialization that
```
Comment (braingram) [inline docs/roman/stpipe/user_step.rst:236]: Please restore this section.

File: docs/roman/references_general/references_general.rst [hunk 18]
```diff
@@ -76,20 +72,12 @@ documentation on each reference file.
 +--------------------------------------------------+---------------------------------------------+
 | :ref:`FLAT <flat_reffile>`                       | :ref:`flatfield <flatfield_step>`           |
 +--------------------------------------------------+---------------------------------------------+
-| :ref:`GAIN <gain_reffile>`                       | :ref:`jump_detection <jump_step>`           |
```
Comment (braingram) [inline docs/roman/references_general/references_general.rst:79]: Please restore this. The `GAIN` reference file is still used by the ramp step.

File: docs/roman/references_general/references_general.rst [hunk 19]
```diff
@@ -76,20 +72,12 @@ documentation on each reference file.
...
-+                                                  +---------------------------------------------+
-|                                                  | :ref:`ramp_fitting <ramp_fitting_step>`     |
-+--------------------------------------------------+---------------------------------------------+
 | :ref:`LINEARITY <linearity_reffile>`             | :ref:`linearity <linearity_step>`           |
 +--------------------------------------------------+---------------------------------------------+
 | :ref:`MASK <mask_reffile>`                       | :ref:`dq_init <dq_init_step>`               |
 +--------------------------------------------------+---------------------------------------------+
 | :ref:`PHOTOM <photom_reffile>`                   | :ref:`photom <photom_step>`                 |
 +--------------------------------------------------+---------------------------------------------+
-| :ref:`READNOISE <readnoise_reffile>`             | :ref:`jump_detection <jump_step>`           |
```
Comment (braingram) [inline docs/roman/references_general/references_general.rst:89]: Same as above with `GAIN`.

**Review by braingram — CHANGES_REQUESTED** [review]
Thanks for working on this. I left a few more requested changes.

**Comment by ddavis-stsci** [comment]
Since there seem to be no more comment I'm merging this. Any issues can be handled in a future ticket.

**Comment by braingram** [comment]
Was this PR approved? I didn't approve it.

File: docs/roman/references_general/references_general.rst (hunk 18 above)
Comment (braingram) [reply docs/roman/references_general/references_general.rst:79]: This is still not correct. The table is not formatted and `GAIN` and `READNOISE` are missing from the lower tables:
https://roman-pipeline--1534.org.readthedocs.build/en/1534/roman/references_general/references_general.html#reference-file-types

Comment (braingram) [reply docs/roman/references_general/references_general.rst:79]: @ddavis-stsci would you address this issue?

**Pull request merged**